src/components/Navbar/NavbarCollapse.tsx

```diff
@@ -18,14 +18,22 @@ export interface NavbarCollapseProps extends ComponentProps<'div'> {
 }
 
 export const NavbarCollapse: FC<NavbarCollapseProps> = ({ children, className, theme: customTheme = {}, ...props }) => {
-  const { theme: rootTheme, isOpen } = useNavbarContext();
+  const { theme: rootTheme, isOpen, setIsOpen } = useNavbarContext();
 
   const theme = mergeDeep(rootTheme.collapse, customTheme);
 
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
   return (
     <div
       data-testid="flowbite-navbar-collapse"
       className={twMerge(theme.base, theme.hidden[!isOpen ? 'on' : 'off'], className)}
+      onClick={handleClick}
+      onKeyDown={handleClick}
+      role="button"
```

Comment:
Wouldnt it be better to add this to the NavBarLink component (it also has access to the navbar context) instead of making the NavBarCollapse an interactive element?

Reply:
Make Sense! Thanks @nigellima . Made the changes accordingly and updated the changes with push